File: app/src/main/java/com/example/weatherapp/MainActivity.java
package com.example.weatherapp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.fragment.app.Fragment;

import android.app.AlarmManager;
import android.app.Notification;
import android.app.PendingIntent;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import com.koushikdutta.async.future.FutureCallback;
import com.koushikdutta.ion.Ion;

import org.jetbrains.annotations.NotNull;

import java.util.Calendar;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
getSupportActionBar().hide();
        BottomNavigationView bottomNav = findViewById(R.id.bottom_navi);
        bottomNav.setOnNavigationItemSelectedListener(selectedLis);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragmect_con,
                new HomeFragment()).commit();
    }
private BottomNavigationView.OnNavigationItemSelectedListener selectedLis =
        new BottomNavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull @NotNull MenuItem item) {
                Fragment selectedFragment = null;
                switch(item.getItemId()){
                    case R.id.nav_drink:
                        selectedFragment = new DrinkFragment();
                        break;
                    case R.id.nav_home:
                        selectedFragment = new HomeFragment();
                        break;
                    case R.id.nav_eat:
                        selectedFragment = new EatFragment();
                        break;
                }
                getSupportFragmentManager().beginTransaction().replace(R.id.fragmect_con,
                        selectedFragment).commit();
                return true;
            }
        };



}

File: app/src/main/java/com/example/whattoeat2/view/MainActivity.kt
package com.example.whattoeat2.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatDelegate
import androidx.fragment.app.Fragment
import com.example.whattoeat2.R
import com.example.whattoeat2.databinding.ActivityMainBinding
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val mainFragment= MainFragment()
        val eatFragment= FoodFragment()
        val drinkFragment= DrinkFragment()
        setCurrentFragment(mainFragment)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)

        bottom_nav_bar.setOnItemSelectedListener {
            when (it) {

                R.id.nav_home -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.main_frag, mainFragment).commit()
                }
                R.id.nav_food -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.main_frag, eatFragment).commit()

                }
                R.id.nav_drink -> {
                    supportFragmentManager.beginTransaction()
                        .replace(R.id.main_frag, drinkFragment).commit()
                    }
            }
        }
    }
    private fun setCurrentFragment(fragment: Fragment)=
        supportFragmentManager.beginTransaction().apply {
            replace(R.id.main_frag,fragment)
            commit()

        }
    }




